### DISCUSSION TITLE
Gatsby Image Plugin Vertical Overflow

### DISCUSSION THREAD

**Discussion by Twads9816** [gatsbyjs/gatsby#36671] [Help]
Hi, I've been struggling with this issue on a personal project and wanted to see if I'm just being stupid. Searching online yields a lot of older gatsby v3 threads, using the older gatsby image solution. Here's basically what I have:
`<section style={{height: "500px"}}>
        <StaticImage
          src="../images/Casual_PFP.jpg"
          alt="Profile picture"
          layout="constrained"
          height="100%"
          objectFit="contain"
        /> </section>`
The image doesn't resize to the height of the parent container, but I would like it to. I don't know if this is by design. In the [documentation](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-plugin-image/#layout) it seems the "constrained" layout responds to the width of its container, but not height. Is there a setting or prop that I'm missing? I would expect gatsby has a non-hacky way to do this but so far I've had no luck. Here's a screenshot of my results, with the parent section highlighted:

![image](https://user-images.githubusercontent.com/37310147/191869043-e50b5a5a-a43b-424c-bc9a-3da5bdb3a2d2.png)

**Comment by LekoArts** [comment]
Why don't you use this?

```js
<StaticImage src="../images/Casual_PFP.jpg" alt="Profile picture" layout="constrained" height="500px" objectFit="contain" />
```

**Reply by Twads9816** [reply]
So I have a div of text in the section, and ideally I would like the section to shrink to the size of the text div, shrinking the image with it. Without the image everything works as expected, but after adding the image (and not enforcing size of the section) the section expands to the height of the image. Perhaps this a situation for flexbox?

![image](https://user-images.githubusercontent.com/37310147/192067503-d509e383-c739-436e-be99-c753b2d7fc87.png)

**Comment by Twads9816** [comment]
So the image resizes in response to a flex or grid container, which works well enough. Now it's more of a question of how to get an element in a container like that to fill in the left over space instead of forcing the container larger.
